File: server_side/haskell-excalibur/excalibur-data/src/ExcaliburData/Transaction.hs
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE DeriveGeneric #-}


module ExcaliburData.Transaction (Transaction(..)) where

import GHC.Generics
import qualified Data.UUID as UUID
import qualified Data.Time.Calendar as Calendar
import Data.Aeson.Types



data Transaction = Transaction {id :: UUID.UUID,
                                date :: Calendar.Day,
                                vendor :: String,
                                amount :: Double}
  deriving (Show, Eq, Generic)


instance FromJSON Transaction
instance ToJSON Transaction

File: server_side/dotnet-excalibur/Excalibur/Models/Transaction.cs
using System;
using System.Collections.Generic;

namespace Excalibur.Models
{
    public partial class Transaction
    {
        public Guid Id { get; set; }
        public DateTime Date { get; set; }
        public string Vendor { get; set; }
        public decimal Amount { get; set; }
    }
}
